### ISSUE TITLE
Tests fail after version update

### ISSUE THREAD

**Issue by DebVortex** [django/channels#368]
Hi,

for my sideprojects to play around with channels, I have quite a good testcoverage and also use a CI System and requires.io to keep my versions up to date. However, the current pullrequests fails and I struggle to find the reason.

The related version changes are:
- asgiref from 0.13.3 to 0.14.0
- channels from 0.16.1 to 0.17.2
- daphne from 0.14.2 to 0.15.0
- Django from 1.9.7 to 1.10.1

In the pullrequest are a few other changes, yet I doubt they are related to the problem.

All tests fail with the message, that a consumer can't be found for the supplied message. Has the `apply_routes` decorator changed lately?

I tried to reproduce the problem with an "empty" project. Here Is was I did:
- created a new virtualenv with python 3
- installed the packages above in the provided versions with `pip install Django==1.10.1 daphne==0.15.0 channels==0.17.2 asgiref==0.14.0`
- created a new project with `django-admin startproject channels_test` and go into the directory
- created a app with `python manage.py startapp foo`
- added `channels` and `foo` to installed apps
- added a `routing.py`, a `tests.py` and a `consumers.py` to the foo app with the content below
- added the following `CHANNEL_LAYERS` setting

in settings.py

```
CHANNEL_LAYERS = {
    "default": {
        "BACKEND": "asgiref.inmemory.ChannelLayer",
        "ROUTING": "foo.routing.channel_routing",
    },
}
```

routing.py

```
from channels.routing import route

from .consumers import ws_connect

channel_routing = [
    route("websocket.connect", ws_connect, path=r"/ws"),
]
```

comsumers.py

```
from channels import Group
from channels.auth import channel_session_user_from_http


@channel_session_user_from_http
def ws_connect(message):
    if message.user.is_authenticated():
        Group('user-{}'.format(message.user.pk)).add(message.reply_channel)
        Group('user-{}'.format(message.user.pk)).send({'text': "Connected!"})
    else:
        message.reply_channel.send({'text': "Not authenticated."})
```

tests.py

```
from channels.tests import ChannelTestCase, apply_routes
from channels.tests.http import HttpClient
from django.contrib.auth import get_user_model

from foo.routing import channel_routing

user_model = get_user_model()


@apply_routes([channel_routing])
class TestWebsocketConsumer(ChannelTestCase):
    """Test the Websocket consumers."""

    def test_ws_connect_not_authenticated(self):
        """Test that ws_connect is not possible if user is not logged in."""
        client = HttpClient()
        client.send_and_consume(u'websocket.connect', {'path': "/ws"})
        response = client.receive()
        self.assertEqual(response, {'text': "Not authenticated."})

    def test_ws_connect_authenticated(self):
        """Test that ws_connect is possible if user is logged in."""
        user_model.objects.create_user('Ada Lovelace', 'user_1@test.test', '123')
        client = HttpClient()
        client.login(username='Ada Lovelace', password='123')
        client.send_and_consume(u'websocket.connect', {'path': "/ws"})
        response = client.receive()
        self.assertEqual(response, {'text': '{"info": "Connected!"}'})
```

Now, when I run the tests I get the following output:

```
$ python manage.py test
Creating test database for alias 'default'...
FF
======================================================================
FAIL: test_ws_connect_authenticated (foo.tests.TestWebsocketConsumer)
Test that ws_connect is possible if user is logged in.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/foo/channels_test/foo/tests.py", line 26, in test_ws_connect_authenticated
    client.send_and_consume(u'websocket.connect', {'path': "/ws"})
  File "/home/max/.virtualenvs/channel-test/lib/python3.4/site-packages/channels/tests/base.py", line 127, in send_and_consume
    return self.consume(channel, fail_on_none=fail_on_none)
  File "/home/max/.virtualenvs/channel-test/lib/python3.4/site-packages/channels/tests/base.py", line 118, in consume
    raise AssertionError("Can't find consumer for message %s" % message)
AssertionError: Can't find consumer for message <channels.message.Message object at 0x7f5029927b38>

======================================================================
FAIL: test_ws_connect_not_authenticated (foo.tests.TestWebsocketConsumer)
Test that ws_connect is not possible if user is not logged in.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/foo/channels_test/foo/tests.py", line 17, in test_ws_connect_not_authenticated
    client.send_and_consume(u'websocket.connect', {'path': "/ws"})
  File "/home/max/.virtualenvs/channel-test/lib/python3.4/site-packages/channels/tests/base.py", line 127, in send_and_consume
    return self.consume(channel, fail_on_none=fail_on_none)
  File "/home/max/.virtualenvs/channel-test/lib/python3.4/site-packages/channels/tests/base.py", line 118, in consume
    raise AssertionError("Can't find consumer for message %s" % message)
AssertionError: Can't find consumer for message <channels.message.Message object at 0x7f5029932b00>

----------------------------------------------------------------------
Ran 2 tests in 0.087s

FAILED (failures=2)
Destroying test database for alias 'default'...
```

What change have I missed? Unfortunately, the test client is not documented, yet.

**Comment by ssteinerx** [comment]
This may be related to a bug I just fixed in another project.  They tightened up the password rules in Django 1.10 and, if the passwords aren't long/strong enough, you will get an error about that instead of the expected one which is what this looks like.

Find where the users are created and give them longer/more complicated passwords.  15 chars, mixed case/number/symbol should do it.

**Comment by ssteinerx** [comment]
Specifically, this `user_model.objects.create_user('Ada Lovelace', 'user_1@test.test', '123')` (Ada should know better!).  The password `123` is invalid and will be rejected.

**Comment by DebVortex** [comment]
Hi @ssteinerx,

thanks for pointing that out. Yet, that is not the problem. As you can see, there is also a test case for a not authenticated user. That results in the same error, that the route for the ``websocket.connect`` channel is not found.

**Comment by Krukov** [comment]
Hi @DebVortex 

Yeah, that is problem at `send` method of `HttpClient` (and I did it). That was already fixed at https://github.com/django/channels/commit/c8fdea646075a55da16b2cba897e7b945b16bac3 but there is not release with this fix.

**Comment by DebVortex** [comment]
Hey @Krukov,

thanks for the reply! So I wait then for the next release and see/hope if that fixes the problem.

**Comment by ssteinerx** [comment]
@Krukov I don't see a fix for this particular issue in  https://github.com/django/channels/commit/c8fdea646075a55da16b2cba897e7b945b16bac3 .

Did it end up in a PR with something that did?

**Comment by Krukov** [comment]
The problem was at ordering of kwargs at `send` method, it was incorrect and different as `send_and_consume` expected(https://github.com/django/channels/commit/c8fdea646075a55da16b2cba897e7b945b16bac3#diff-e0edf374a5a1aee9327af30e22121a6cR64).
 Related PR https://github.com/django/channels/pull/320

**Comment by Krukov** [comment]
@andrewgodwin @DebVortex 
Can we close this issue? Fixed at 0.17.3 release.

**Comment by DebVortex** [comment]
@Krukov @andrewgodwin sorry that I forgot to response. Yes, the Issue is fixed. Thanks! :)
